### PR TITLE
Challenge well formed strings

### PR DESCRIPTION
**Pull request by vidoseaver**

### CODE REVIEW

**Comment by jcasimir** [comment]
👍

File: well_formed_strings.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    string.chars.each do |char|
+      if char == "("
+        paren_open == true
+        paren_count = paren_count + 1
+      elsif char ==")" && paren_open == true
+        paren_count = paren_count - 1
+        paren_open = false
+      end
+    end
+       paren_count == 0 && paren_open == false ? !paren_open : paren_open
```
Comment (mikedao) [inline well_formed_strings.rb:16]: I'd probably pull out this conditional to a predicate method.

File: well_formed_strings_test.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+gem 'minitest', '~> 5.2'
+require 'minitest/autorun'
+require 'minitest/pride'
+require_relative 'well_formed_strings'
+
```
Comment (mikedao) [inline well_formed_strings_test.rb:5]: Would love to see more sad path and edge case testing here!

**Review by mikedao — APPROVED** [review]

File: well_formed_strings_test.rb (hunk 2 above)
Comment (vidoseaver) [reply well_formed_strings_test.rb:5]: gotchya, thanks for the review

**Pull request closed without merging**